cordapp-upgrades/README.md

```diff
@@ -0,0 +1,158 @@
+ This creates a network of 3 nodes, all running version 1 of the contract. PartyA and PartyB are using the original
+ version of the workflows, while PartyC is using version 2.
+ 
+ 2. Check that all the nodes can transact with each other. This can be done manually from the shell, or by running
+ `./gradlew issueBetweenNodes`. This will issue an obligation between each pair of nodes in the network.
+ 
+ 3. Now shut down the nodes and upgrade all nodes to version 2 of the workflows CorDapp. The upgrade part of this can be
+ carried out by running `scripts/upgradeNodes.sh workflows v2-new-finality-flow PartyA PartyB`.
+ This simply copies the workflow jars (and any additional configuration) to the cordapps/ directory on the nodes specified.
+ 
+ 4. Restart the nodes using `build/nodes/runnodes`. All nodes should now be running V2 of the workflows CorDapp. The nodes
+ should still be able to transact with each other. New obligations can again be issued between all nodes using 
+ `./gradlew issueBetweenNodes`, and old obligations can be settled using `./gradlew settleAllObligations`. The shell can
+ also be used to experiment.
+ 
+Note: If this example is compiled against Corda 4.0, then a bug will prevent this scenario from executing successfully.
+The issue is that the new version of `FinalityFlow` in 4.0 has a `targetPlatformVersion` gate that prevents the old API
+from being called if the app has set `targetPlatformVersion = 4`. To fix this, change the V2 workflows app to use
+`targetPlatformVersion = 3` instead. This issue is fixed in Corda 4.1.
+
+## Scenario 2: Contract Version Upgrade
+
+This section describes how to carry out an implicit contract upgrade. The upgraded contract in this case adds a new
+field to the obligation state (a defaulted flag), and a new command to govern transactions that set it. Additionally,
+version 4 of the workflows CorDapp provides a flow that can be used to set this flag.
+
+1. Deploy the nodes in their initial configuration by running `./gradlew deployNodes`. Start the network by
+ running `build/nodes/runnodes` and observe that all nodes can transact with each other (this can be done by running
+ `./gradlew issueBetweenNodes`, which will issue an obligation between each pair of nodes in the network).
+ 
+2. Shut down the network, then upgrade one of the nodes to the new version of the contract. This also requires
+ upgrading to the latest version of the workflows CorDapp, as older versions are compiled against the initial contract.
+ The upgrade can be carried out by running `scripts/upgradeNodes.sh workflows v3-can-default PartyA`
+ and `scripts/upgradeNodes.sh contracts v2-can-default PartyA`. Note: There is some additional configuration for the nodes running the new versions of the workflows CorDapp. This
+ configuration is used to control whether the new field in the contract state is set when the state is created. The no
```

Comment:
It's not clear what configuration you're referring to here.

Reply:
ditto - please elaborate.

Reply:
This was some cordapp configuration that is now removed.